### ISSUE TITLE
Crash of iterm with the source code pro, both normal and mono

### ISSUE THREAD

**Issue by hanoii** [ryanoasis/nerd-fonts#258]
I can't seem to make this work, every time I change the iterm settings to use this font iTerm simply starts loading forever. I have to force quit it, and on load it happens again.

The only way to restore it to normal is deleting the settings:

`defaults delete com.googlecode.iterm2`

#### 🔧 Your Setup

OSX 10.13.4
iTerm2 3.1.5
`brew cask install brew cask install font-sourcecodepro-nerd-font`
and
`brew cask install font-sourcecodepro-nerd-font-mono`

Both does the same.

It works with the Hack one for instance.

**Comment by ryanoasis** [comment]
Thanks for the report. Does sourcecodepro (unpatched) work just fine for you? I assume so..

**Comment by pitkling** [comment]
Same problem here. Note that the font does work in other environments (e.g., Vimr).

**Comment by omares** [comment]
I know this is not helping but i came here to report the same issue, as soon as i select the source code pro font my iterm hangs forever.

To unstuck iterm open up the normal mac terminal and remove the font. In case you obtained the font via brew: `brew cask uninstall font-sourcecodepro-nerd-font` will do the job.

**Comment by ryanoasis** [comment]
@snachx @pitkling Thanks for chiming in, I actually appreciate it.

@omares Thanks to you as well for the report and the tip. No worries there is certainly a problem.

Appreciate the patience overall :smile:

**Comment by vheon** [comment]
I've encountered this as well. Is there any way I can provide more data?

**Comment by jerr0328** [comment]
Hey folks, I had the same issue. I noticed in Font Book that the font had multiple versions. I clicked the "resolve automatically" button and then was able to switch to the font in iTerm2.

**Comment by sagikazarmark** [comment]
I usually prefer having these settings scripted so that it is documented and can be automated in case of moving to a new computer, so here it is:

```bash
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Term%20Thin%20Italic%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Italic"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Bold%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Bold"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Term%20Medium%20Oblique%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Oblique"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Medium%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Medium"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Term%20Light%20Oblique%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Oblique"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Thin%20Italic%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Italic"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Term%20Heavy%20Oblique%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Oblique"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Term%20Heavy%20Italic%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Italic"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Term%20Thin%20Oblique%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Oblique"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Term%20Bold%20Oblique%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Oblique"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Heavy%20Italic%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Italic"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Light%20Italic%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Italic"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Extralight%20Italic%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Italic"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Extralight%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Extralight"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Thin%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Thin"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Medium%20Oblique%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Oblique"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Medium%20Italic%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Italic"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Term%20Extralight%20Italic%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Italic"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Thin%20Oblique%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Oblique"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Bold%20Oblique%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Oblique"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Term%20Oblique%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Oblique"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Term%20Light%20Italic%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Italic"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Italic%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Italic"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Light%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Light"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Heavy%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Heavy"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Term%20Medium%20Italic%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Italic"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Oblique%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Oblique"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Extralight%20Oblique%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Oblique"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Light%20Oblique%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Oblique"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Bold%20Italic%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Bold-Italic"
defaults write com.apple.FontRegistry.user DisabledFonts -array-add "file://$HOME/Library/Fonts/Iosevka%20Heavy%20Oblique%20Nerd%20Font%20Complete.ttf#postscript-name=IosevkaNerdFontComplete-Oblique"
```

**Comment by Finii** [comment]
> I noticed in Font Book that the font had multiple versions.

These will all be fixed with `v3.0.0` with the new naming engine.

**Comment by Finii** [comment]
Release is in repo, release as packages pending.
